cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -77,6 +77,7 @@ add_arrow_lib(arrow_flight
               ${ARROW_FLIGHT_SRCS}
               DEPENDENCIES
               flight_grpc_gen
+              arrow_dependencies
```

Comment:
Hm, it seems like the missing dependency is `metadata_fbs`, is that right?

Reply:
you are right, fixed it.